tests/unit/s2n_certificate_test.c

```diff
@@ -19,6 +19,53 @@
 #include <s2n.h>
 #include "utils/s2n_safety.h"
 
+#include <openssl/crypto.h>
+#if defined(OPENSSL_IS_AWSLC)
+#include <openssl/mem.h>
+#endif
+
+struct host_verify_data {
+    const char *name;
+    uint8_t found_name;
+    uint8_t callback_invoked;
+};
+
+uint32_t write_pem_file_to_stuffer_as_chain(struct s2n_stuffer *chain_out_stuffer, const char *pem_data, uint8_t protocol_version) {
```

Comment:
This function doesn't `GUARD` any errors at all. This will especially cause issues when any of the calling functions are changed to return a `S2N_RESULT` since it enforces `GUARD`ing. Change the return type to `S2N_RESULT` and move the `chain_size` to an out param

Reply:
This is supposed to be a static helper function, we probably want to put this in testlib as it's used in multiple places within tests/unit and tests/fuzz. I will add the `GUARD` and change it to return a `S2N_RESULT`.

Reply:
> This is supposed to be a static helper function

That's fine; we just want to make sure we're handling errors in tests as well to improve test confidence